lib/compilers/csharp.js

```diff
@@ -0,0 +1,155 @@
+// Copyright (c) 2021, Compiler Explorer Authors
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+//     * Redistributions of source code must retain the above copyright notice,
+//       this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+import path from 'path';
+
+import fs from 'fs-extra';
+
+import { BaseCompiler } from '../base-compiler';
+
+export class CSharpCompiler extends BaseCompiler {
+
+    static get key() { return 'csharp'; }
+
+    get rID() { return 'linux-x64'; }
+    get targetFramework() { return 'net6.0'; }
+    get buildConfig() { return 'Release'; }
+
+    get compilerOptions() {
+        return ['publish', '-c', this.buildConfig, '--self-contained', '--runtime', this.rID];
+    }
+
+    async runCompiler(compiler, options, inputFileName, execOptions) {
+        if (!execOptions) {
+            execOptions = this.getDefaultExecOptions();
+        }
+
+        const programDir = path.dirname(inputFileName);
+        const clrBuildDir = '/home/ubuntu/runtime/artifacts/bin/coreclr/Linux.x64.Checked';
```

Comment:
this should be specified in the compiler props file as something like:
```
compiler.csharp.clrDir=/home/ubuntu/runtime/artifacts/bin/coreclr/Linux.x64.Checked
```

and then accessed here (or assigned to `this.clrBuildDir` in the class constructor) as:
```
this.compilerProps('clrDir')
```